content/articles/creating-a-honeypot-for-network-security/index.md

```diff
@@ -0,0 +1,139 @@
+author: ruth-ngonyo
+date: 2022-02-10T00:00:00-14:00
+topics: [Security]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/creating-a-honeypot-for-network-security/hero.png
+    alt: Creating a honeypot for network security
+---
+Many companies rely on publicly available data to make critical choices in today's data-driven world. Web scraping is a common technique for obtaining this information from the Internet.
+<!--more-->
+However, web scraping has specific difficulties. The honeypot is an example of a trap. One of the most effective ways to deter hackers is to use honeypots. A honeypot system deters attackers from their objectives. Unwanted behavior is often detected with the use of honeypots by security teams.
+
+This article will discuss honeypots and how to avoid them when scraping the web for data.
+
+### Table of contents
+- [Types of honeypots](#types-of-honeypots)
+- [How honeypots operate](#how-honeypots-operate)
+- [Merits and demerits](#merits-and-demerits)
+- [Honeypot traps and web scraping](#honeypot-traps-and-web-scraping)
+- [Avoiding honeypot traps](#avoiding-honeypot-traps)
+- [Creating a honeypot](#creating-a-honeypot)
+- [Honeypot software](#honeypot-software)
+- [Conclusion](#conclusion)
+
+### Types of honeypots
+Honeypots are classified into the following:
+
+1. Research honeypots
+2. Production honeypots
+
+They are described below.
+
+- `Research honeypots:` Researchers use research honeypots to learn more about the black hat group's methods and goals. These honeypots are used to analyze corporate threats and learn how to mitigate them. Academic honeypots are used by research, military, and government groups to collect large amounts of data.
+
+- `Production honeypots:` Production honeypots are famous for their low data collection and ease of usage. Adding production honeypots improves network security. Due to their little participation, production honeypots are easy to set up. Research honeypots provide more about the assaults and attackers.
+
+Design-wise, honeypots are classed as follows:
+ 
+- Low-interaction honeypots
+- pure honeypots
+- high-interaction honeypots
+
+`Low-interaction honeypots`- replicate attacker-requested services. Multiple virtual machines may be readily hosted on one physical system, and less code is needed, decreasing the virtual system's security complexity. Honey
+
+`Pure honeypots`- are full-scale factories. The honeypot's network connection is bugged to monitor the attacker's activities. A separate piece of software isn't needed. Both pure honeypots and honeypot-like systems have benefits and downsides.
+
+High-interaction honeypots enable an attacker to waste time by mimicking production system activities. Using virtual PCs, a single server may host several honeypots. A quick honeypot penetration may restore it. While high-interaction honeypots are safer, they need more maintenance. It may be too pricey without virtual computers.

```

Comment:
```suggestion
`High-interaction honeypots` enable an attacker to waste time by mimicking production system activities. Using virtual PCs, a single server may host several honeypots. A quick honeypot penetration may restore it. While high-interaction honeypots are safer, they need more maintenance. It may be too pricey without virtual computers.
```